src/utilities/export-pdf-helper.ts

```diff
@@ -0,0 +1,242 @@
+            await question.grades?.sequentialAsyncForEach(async (grade) => {
+                if (_.isNil(studentGradeLookup[grade.userId])) {
+                    studentGradeLookup[grade.userId] = [];
+                }
+
+                const influencingWorkbook = grade.lastInfluencingCreditedAttemptId ?? grade.lastInfluencingAttemptId;
+
+                if (_.isNil(influencingWorkbook)) {
+                    // If we can't find a workbook, we can't display anything for the student attempt.
+                    studentGradeLookup[grade.userId].push({
+                        number: question.problemNumber,
+                        attachments: [],
+                        srcdoc: 'There is no record of this student attempting this problem.',
+                        effectiveScore: grade.effectiveScore,
+                        legalScore: grade.legalScore,
+                    });
+
+                    return;
+                }
+
+                const gradeInstanceAttachments = await StudentWorkbook.findOne({
+                    where: {
+                        id: influencingWorkbook,
+                        active: true,
+                    },
+                    attributes: ['id', 'submitted', 'randomSeed'],
+                    include: [
+                        {
+                            model: StudentGradeInstance,
+                            as: 'studentGradeInstance',
+                            attributes: ['id', 'webworkQuestionPath', 'randomSeed'],
+                            required: false,
+                            where: {
+                                active: true,
+                            },
+                            include: [
+                                {
+                                    model: ProblemAttachment,
+                                    as: 'problemAttachments',
+                                    attributes: ['id', 'cloudFilename', 'userLocalFilename', 'updatedAt'],
+                                    where: {
+                                        active: true,
+                                    }
+                                }
+                            ]
+                        },
+                    ]
+                });
+
+                const obj = {
+                    sourceFilePath: gradeInstanceAttachments?.studentGradeInstance?.webworkQuestionPath ?? question.webworkQuestionPath,
+                    problemSeed: gradeInstanceAttachments?.randomSeed ?? gradeInstanceAttachments?.studentGradeInstance?.randomSeed ?? grade.randomSeed,
+                    formData: gradeInstanceAttachments?.submitted.form_data,
+                    showCorrectAnswers: true,
+                    answersSubmitted: 1,
+                    outputformat: rendererHelper.getOutputFormatForRole(Role.PROFESSOR),
+                    permissionLevel: rendererHelper.getPermissionForRole(Role.PROFESSOR),
+                    showSolutions: true,
```

Comment:
Does show solutions work since they are collapsible?

Reply:
Hints and solutions are obscured as 'knowls'
They are wrapped in an <a> tag, with class 'internal' 
The addition of 'active' to the class causes the hint/solution to be rendered on the page.